app/src/main/java/com/swentseekr/seekr/ui/profile/EditProfileScreen.kt

```
@@ -101,24 +140,25 @@ fun EditProfileScreen(
 
   fun launchCamera() {
     try {
-      cameraPhotoUri = createTempImageUri()
-      cameraLauncher.launch(cameraPhotoUri)
+      val uri = createImageUri(context)
```

createImageUri uses !! – if the insert fails, you’ll throw NPE. You’re catching Exception, so you won’t crash, but you’ll get a log and no feedback in the UI. Might be nicer to:

Handle a null URI and show an error message via the ViewModel instead of silently failing.